miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -349,6 +365,24 @@ export.get_storage_slot_type_info
     exec.memory::get_acct_storage_slot_type_data u32split
 end
 
+#! Offsets provided storage slot index for storage access
+#!
+#! Panics:
+#!   - slot index is out of bounds
+#!   - computed index is out of bounds
+#!
+#! Stack: [storage_offset, slot_index]
+#! Output: [offset_slot_index]
+export.offset_storage_index
```

Comment:
nit: I would rename this to `apply_storage_offset` or something like this (`offset` verb can be confused with `offset` noun pretty easily).